test/vscode-ui/test/specs/extension.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { browser, $, expect } from "@wdio/globals";
+
+import * as fs from "fs";
+import * as path from "path";
+import { fileURLToPath } from "url";
+import { dirname } from "path";
+
+async function switchToSubframe() {
+  await browser.$(".webview");
+  const iframe = await browser.$("iframe");
+  await browser.switchToFrame(iframe);
+
+  await browser.$("iframe").waitForExist({ timeout: 30000 });
+  const subiframe = await browser.$("iframe");
+  await subiframe.waitForExist({ timeout: 30000 });
+  await browser.switchToFrame(subiframe);
+}
+
+function generateRandomString(length = 10) {
+  const characters =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  let result = "";
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length));
+  }
+  return result;
+}
+
+const serverName = generateRandomString();
+const apiKey = generateRandomString(32);
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+describe("VS Code Extension UI Test", () => {
+  let workbench: any;
+  let input: any;
+
+  before(async () => {
+    workbench = await browser.getWorkbench();
+    input = await $(".input");
+  });
+
+  it("open extension", async () => {
+    browser.$("aria/Posit Publisher").waitForExist({ timeout: 30000 });
```

Comment:
What is your approach for the selection criteria's you are using? It would be helpful to identify attributes you can count on not being modified or lost (at least we can be aware of the usage), so that the tests won't need to be modified as much with code changes in the underlying interfaces. Have you just been looking what is set now? Should we start putting a `data-` attribute on the web components  that you may need to access?

Reply:
> What is your approach for the selection criteria's you are using? It would be helpful to identify attributes you can count on not being modified or lost (at least we can be aware of the usage), so that the tests won't need to be modified as much with code changes in the underlying interfaces. Have you just been looking what is set now? Should we start putting a data- attribute on the web components that you may need to access?

`data-` attributes would be great.  I was following the best practices for webdriver here, they say the aria labels should be used for selectors. 
https://webdriver.io/docs/selectors

If we have `data` selectors I think that would be preferable though. I can certainly add them but was having a hard time finding where they should go.

> Those selectors can be addressed as we move forward, as long as these tests are not going to block our CI when they fail?

Right now, they'd be running every in CI on every PR. I can make them run only at night though.